### ISSUE TITLE
icinga2 inputs module not reporting affected asset hostname

### ISSUE THREAD

**Issue by buggsdummy** [influxdata/telegraf#7004]
### Relevant telegraf.conf:

```
[[inputs.icinga2]]
  ## Required Icinga2 server address (default: "https://localhost:5665")
  server = "https://localhost:5665"
  
  ## Required Icinga2 object type ("services" or "hosts, default "services")
  object_type = "services"

  ## Credentials for basic HTTP authentication
  username = "root"
  password = "abc123"

  ## Maximum time to receive response.
  response_timeout = "5s"

  ## Optional TLS Config
  # tls_ca = "/etc/telegraf/ca.pem"
  # tls_cert = "/etc/telegraf/cert.pem"
  # tls_key = "/etc/telegraf/key.pem"
  ## Use TLS but skip chain & host verification
  insecure_skip_verify = true

```

### System info:

telegraf-1.12.1-1.x86_64

```
CentOS Linux release 7.7.1908 (Core)
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

CentOS Linux release 7.7.1908 (Core)
CentOS Linux release 7.7.1908 (Core)
```

### Steps to reproduce:

```
    1. Check the icinga2 api for a check with a notOK state
    https://localhost:5665/v1/objects/services?filter=service.state!=ServiceOK

    2. Check the metrics on icinga2 server
    http://localhost:9126/metrics
```

### Expected behavior:

Telegraf should output the affected hostname from the icinga2 api such as the appserver

### Actual behavior:

Telegraf is missing the output of the affected hostname

### Additional info:

http://icinga2server.net:9126/metrics
```
icinga2_services_state_code{check_command="daily_processor_completed_check_me",display_name="daily_processor_completed_check_me",host="icinga2server.net",name="daily_processor_completed_check_me",port="5665",scheme="https",source="localhost",state="critical"} 2

```

icinga2 API
```

1 |  
-- | --
attrs |  
__name | "appserver.net!daily_processor_completed_check_me"
acknowledgement | 0
acknowledgement_expiry | 0
action_url | ""
active | true
check_attempt | 1
check_command | "daily_processor_completed_check_me"
check_interval | 300
check_period | "twentyfour x seven"
check_timeout | null
command_endpoint | ""
display_name | "daily_processor_completed_check_me"
downtime_depth | 0
enable_active_checks | false
enable_event_handler | true
enable_flapping | false
enable_notifications | true
enable_passive_checks | true
enable_perfdata | true
event_command | ""
flapping | false
flapping_current | 5.9
flapping_last_change | 0
flapping_threshold | 0
flapping_threshold_high | 30
flapping_threshold_low | 25
force_next_check | false
force_next_notification | false
groups | []
ha_mode | 0
handled | false
host_name | "appserver.net"
icon_image | ""
icon_image_alt | ""
last_check | 1581359430
last_check_result |  
active | false
check_source | "icinga2server.net"
command | null
execution_end | 1581359430
execution_start | 1581359430
exit_status | 0
output | "Not OK"
performance_data | []
schedule_end | 1581359430
schedule_start | 1581359430
state | 2
ttl | 0
type | "CheckResult"
vars_after |  
attempt | 1
reachable | true
state | 2
state_type | 0
vars_before |  
attempt | 1
reachable | true
state | 0
state_type | 1
last_hard_state | 0
last_hard_state_change | 1581001272.568679
last_reachable | true
last_state | 0
last_state_change | 1581359430.435301
last_state_critical | 1581359430.435313
last_state_ok | 1581359112.562763
last_state_type | 1
last_state_unknown | 1574774178.149394
last_state_unreachable | 1574773888.107177
last_state_warning | 1581001212.528714
max_check_attempts | 3
name | "daily_processor_completed_check_me"
next_check | 1581359730.435331
notes | ""
notes_url | ""
original_attributes |  
enable_active_checks | true
package | "director"
paused | false
previous_state_change | 1581001272.568679
problem | true
retry_interval | 60
severity | 136
source_location |  
first_column | 1
first_line | 281
last_column | 56
last_line | 281
path | "/var/lib/icinga2/api/packages/director/7ef8d9fd-ce33-455d-804c-f06b4937555d/zones.d/director-global/servicesets.conf"
state | 2
state_type | 0
templates |  
0 | "daily_processor_completed_check_me"
1 | "daily_processor_completed_check_me"
2 | "host var overrides (Director)"
type | "Service"
vars | null
version | 1581359408.21923
volatile | false
zone | "master"
joins | {}
meta | {}
name | "appserver.net!daily_processor_completed_check_me"
type | "Service"

```

**Comment by danielnelson** [comment]
Can you run this and add the output?
```
curl --user root:abc123 'http://localhost:5665/v1/objects/services?attrs=name&attrs=display_name&attrs=state&attrs=check_command'
```

**Comment by buggsdummy** [comment]
```
{"results":[{"attrs":{"check_command":"Check DB backup","display_name":"Check DB backup","name":"Check DB backup","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB backup","type":"Service"},{"attrs":{"check_command":"load","display_name":"load","name":"load","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!load","type":"Service"},{"attrs":{"check_command":"disk","display_name":"disk /","name":"disk /","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!disk /","type":"Service"},{"attrs":{"check_command":"icinga","display_name":"icinga","name":"icinga","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!icinga","type":"Service"},{"attrs":{"check_command":"check_dr_processor_idmissing_reference","display_name":"check_dr_processor_idmissing_reference","name":"check_dr_processor_idmissing_reference","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_dr_processor_idmissing_reference","type":"Service"},{"attrs":{"check_command":"ping4","display_name":"ping4","name":"ping4","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!ping4","type":"Service"},{"attrs":{"check_command":"app SFTP Companion File check","display_name":"app SFTP Companion File check","name":"app SFTP Companion File check","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!app SFTP Companion File check","type":"Service"},{"attrs":{"check_command":"check_app_to_process_for_older_files","display_name":"check_app_to_process_for_older_files","name":"check_app_to_process_for_older_files","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_app_to_process_for_older_files","type":"Service"},{"attrs":{"check_command":"ping4","display_name":"ping4","name":"ping4","state":0.0},"joins":{},"meta":{},"name":"appserver.net!ping4","type":"Service"},{"attrs":{"check_command":"nrpe-mem","display_name":"check_mem","name":"check_mem","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_mem","type":"Service"},{"attrs":{"check_command":"ping4","display_name":"ping4","name":"ping4","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!ping4","type":"Service"},{"attrs":{"check_command":"daily_processor_completed_check_app","display_name":"daily_processor_completed_check_app","name":"daily_processor_completed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_processor_completed_check_app","type":"Service"},{"attrs":{"check_command":"check_app_idmissing_creative","display_name":"check_app_idmissing_creative","name":"check_app_idmissing_creative","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_app_idmissing_creative","type":"Service"},{"attrs":{"check_command":"app_delivery_stuck_check_app","display_name":"app_delivery_stuck_check_app","name":"app_delivery_stuck_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_stuck_check_app","type":"Service"},{"attrs":{"check_command":"load","display_name":"check_load","name":"check_load","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_load","type":"Service"},{"attrs":{"check_command":"procs","display_name":"procs","name":"procs","state":1.0},"joins":{},"meta":{},"name":"icinga2server.net!procs","type":"Service"},{"attrs":{"check_command":"Processor_failed_check_app","display_name":"Processor_failed_check_app","name":"Processor_failed_check_app","state":0.0},"joins":{},"meta":{},"name":"appserver.net!Processor_failed_check_app","type":"Service"},{"attrs":{"check_command":"check_app_brand_pop_map_error","display_name":"check_app_brand_pop_map_error","name":"check_app_brand_pop_map_error","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_app_brand_pop_map_error","type":"Service"},{"attrs":{"check_command":"load","display_name":"check_load","name":"check_load","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_load","type":"Service"},{"attrs":{"check_command":"daily_processor_completed_check_app","display_name":"daily_processor_completed_check_app","name":"daily_processor_completed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_processor_completed_check_app","type":"Service"},{"attrs":{"check_command":"daily_REFRESH_UI_COMPLETED_check","display_name":"daily_REFRESH_UI_COMPLETED_check","name":"daily_REFRESH_UI_COMPLETED_check","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_REFRESH_UI_COMPLETED_check","type":"Service"},{"attrs":{"check_command":"disk","display_name":"disk","name":"disk","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!disk","type":"Service"},{"attrs":{"check_command":"check_dr_processor_idmissing_impression","display_name":"check_dr_processor_idmissing_impression","name":"check_dr_processor_idmissing_impression","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_dr_processor_idmissing_impression","type":"Service"},{"attrs":{"check_command":"check_app_files","display_name":"check_app_files","name":"check_app_files","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_app_files","type":"Service"},{"attrs":{"check_command":"ping6","display_name":"ping6","name":"ping6","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!ping6","type":"Service"},{"attrs":{"check_command":"ping4","display_name":"ping4","name":"ping4","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!ping4","type":"Service"},{"attrs":{"check_command":"daily_app_delivery_check_app","display_name":"daily_app_delivery_check_app","name":"daily_app_delivery_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_app_delivery_check_app","type":"Service"},{"attrs":{"check_command":"check_app_processor_daily","display_name":"check_app_processor_daily","name":"check_app_processor_daily","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_app_processor_daily","type":"Service"},{"attrs":{"check_command":"daily_impressions_check_app","display_name":"daily_impressions_check_app","name":"daily_impressions_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_impressions_check_app","type":"Service"},{"attrs":{"check_command":"users","display_name":"users","name":"users","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!users","type":"Service"},{"attrs":{"check_command":"ping4","display_name":"ping4","name":"ping4","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!ping4","type":"Service"},{"attrs":{"check_command":"check_app_processor_pop","display_name":"check_app_processor_pop","name":"check_app_processor_pop","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_app_processor_pop","type":"Service"},{"attrs":{"check_command":"daily_app_delivery_check_app","display_name":"daily_app_delivery_check_app","name":"daily_app_delivery_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_app_delivery_check_app","type":"Service"},{"attrs":{"check_command":"daily_impressions_check_app","display_name":"app_delivery_stuck_check_app","name":"app_delivery_stuck_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_stuck_check_app","type":"Service"},{"attrs":{"check_command":"disk","display_name":"check_disk","name":"check_disk","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_disk","type":"Service"},{"attrs":{"check_command":"check_app_cal","display_name":"check_app_cal","name":"check_app_cal","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_app_cal","type":"Service"},{"attrs":{"check_command":"app_delivery_failed_check_app","display_name":"app_delivery_failed_check_app","name":"app_delivery_failed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_failed_check_app","type":"Service"},{"attrs":{"check_command":"nrpe-mem","display_name":"check_mem","name":"check_mem","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_mem","type":"Service"},{"attrs":{"check_command":"app_delivery_stuck_check_app","display_name":"app_delivery_stuck_check_app","name":"app_delivery_stuck_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_stuck_check_app","type":"Service"},{"attrs":{"check_command":"app_delivery_stuck_check_app","display_name":"app_delivery_stuck_check_app","name":"app_delivery_stuck_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_stuck_check_app","type":"Service"},{"attrs":{"check_command":"check_app_time_out","display_name":"check_app_time_out","name":"check_app_time_out","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_app_time_out","type":"Service"},{"attrs":{"check_command":"Check DB","display_name":"Check DB","name":"Check DB","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB","type":"Service"},{"attrs":{"check_command":"load","display_name":"check_load","name":"check_load","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_load","type":"Service"},{"attrs":{"check_command":"check_app_unmap_device","display_name":"check_app_unmap_device","name":"check_app_unmap_device","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_app_unmap_device","type":"Service"},{"attrs":{"check_command":"app_delivery_failed_check_app","display_name":"app_delivery_failed_check_app","name":"app_delivery_failed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_failed_check_app","type":"Service"},{"attrs":{"check_command":"check db connections","display_name":"Check DB connections","name":"Check DB connections","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB connections","type":"Service"},{"attrs":{"check_command":"ssh","display_name":"check_ssh","name":"check_ssh","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_ssh","type":"Service"},{"attrs":{"check_command":"app Reference File Age - Day","display_name":"app Reference File Age - Day","name":"app Reference File Age - Day","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!app Reference File Age - Day","type":"Service"},{"attrs":{"check_command":"daily_impressions_check_app","display_name":"daily_impressions_check_app","name":"daily_impressions_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_impressions_check_app","type":"Service"},{"attrs":{"check_command":"daily_processor_completed_check_app","display_name":"daily_processor_completed_check_app","name":"daily_processor_completed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_processor_completed_check_app","type":"Service"},{"attrs":{"check_command":"check_app_map_process","display_name":"check_app_map_process","name":"check_app_map_process","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_app_map_process","type":"Service"},{"attrs":{"check_command":"nrpe-mem","display_name":"check_mem","name":"check_mem","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_mem","type":"Service"},{"attrs":{"check_command":"disk","display_name":"check_disk","name":"check_disk","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_disk","type":"Service"},{"attrs":{"check_command":"iZone Scheduler log error check","display_name":"iZone Scheduler log error check","name":"iZone Scheduler log error check","state":0.0},"joins":{},"meta":{},"name":"appserver.net!iZone Scheduler log error check","type":"Service"},{"attrs":{"check_command":"daily_app_delivery_check_app","display_name":"daily_app_delivery_check_app","name":"daily_app_delivery_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_app_delivery_check_app","type":"Service"},{"attrs":{"check_command":"app SFTP 2","display_name":"app SFTP 2","name":"app SFTP 2","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!app SFTP 2","type":"Service"},{"attrs":{"check_command":"check_vid_pop_map_error","display_name":"check_vid_pop_map_error","name":"check_vid_pop_map_error","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_vid_pop_map_error","type":"Service"},{"attrs":{"check_command":"check app to_process for  older files ","display_name":"check app to_process for  older files ","name":"check app to_process for  older files ","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check app to_process for  older files ","type":"Service"},{"attrs":{"check_command":"load","display_name":"check_load","name":"check_load","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_load","type":"Service"},{"attrs":{"check_command":"app_delivery_failed_check_app","display_name":"app_delivery_failed_check_app","name":"app_delivery_failed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_failed_check_app","type":"Service"},{"attrs":{"check_command":"nrpe-mem","display_name":"check_mem","name":"check_mem","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_mem","type":"Service"},{"attrs":{"check_command":"daily_processor_completed_check_app","display_name":"daily_processor_completed_check_app","name":"daily_processor_completed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_processor_completed_check_app","type":"Service"},{"attrs":{"check_command":"daily_app_delivery_check_app","display_name":"daily_app_delivery_check_app","name":"daily_app_delivery_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_app_delivery_check_app","type":"Service"},{"attrs":{"check_command":"Check DB pg_xlog size","display_name":"Check DB pg_xlog size","name":"Check DB pg_xlog size","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB pg_xlog size","type":"Service"},{"attrs":{"check_command":"daily_impressions_check_app","display_name":"app_delivery_failed_check_app","name":"app_delivery_failed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_failed_check_app","type":"Service"},{"attrs":{"check_command":"disk","display_name":"check_disk","name":"check_disk","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_disk","type":"Service"},{"attrs":{"check_command":"app_delivery_failed_check_app","display_name":"app_delivery_failed_check_app","name":"app_delivery_failed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_failed_check_app","type":"Service"},{"attrs":{"check_command":"iverify sftp mount check","display_name":"iverify sftp mount check","name":"iverify sftp mount check","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!iverify sftp mount check","type":"Service"},{"attrs":{"check_command":"ssh","display_name":"ssh","name":"ssh","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!ssh","type":"Service"},{"attrs":{"check_command":"Check DB","display_name":"Check DB","name":"Check DB","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB","type":"Service"},{"attrs":{"check_command":"check_app_pop_map_error","display_name":"check_app_pop_map_error","name":"check_app_pop_map_error","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_app_pop_map_error","type":"Service"},{"attrs":{"check_command":"app_delivery_stuck_check_app","display_name":"app_delivery_stuck_check_app","name":"app_delivery_stuck_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!app_delivery_stuck_check_app","type":"Service"},{"attrs":{"check_command":"check_app_platform_pop_map_error","display_name":"check_app_platform_pop_map_error","name":"check_app_platform_pop_map_error","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_app_platform_pop_map_error","type":"Service"},{"attrs":{"check_command":"swap","display_name":"swap","name":"swap","state":0.0},"joins":{},"meta":{},"name":"icinga2server.net!swap","type":"Service"},{"attrs":{"check_command":"nrpe-mem","display_name":"check_mem","name":"check_mem","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_mem","type":"Service"},{"attrs":{"check_command":"ssh","display_name":"check_ssh","name":"check_ssh","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_ssh","type":"Service"},{"attrs":{"check_command":"daily_processor_completed_check_app","display_name":"daily_processor_completed_check_app","name":"daily_processor_completed_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_processor_completed_check_app","type":"Service"},{"attrs":{"check_command":"check_app_idmissing_sli","display_name":"check_app_idmissing_sli","name":"check_app_idmissing_sli","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_app_idmissing_sli","type":"Service"},{"attrs":{"check_command":"daily_app_delivery_check_app","display_name":"daily_app_delivery_check_app","name":"daily_app_delivery_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_app_delivery_check_app","type":"Service"},{"attrs":{"check_command":"ssh","display_name":"check_ssh","name":"check_ssh","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_ssh","type":"Service"},{"attrs":{"check_command":"check_app_broadcast","display_name":"check_app_broadcast","name":"check_app_broadcast","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_app_broadcast","type":"Service"},{"attrs":{"check_command":"check_app","display_name":"check_app","name":"check_app","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_app","type":"Service"},{"attrs":{"check_command":"load","display_name":"check_load","name":"check_load","state":0.0},"joins":{},"meta":{},"name":"appserver.net!check_load","type":"Service"},{"attrs":{"check_command":"daily_impressions_check_app","display_name":"daily_impressions_check_app","name":"daily_impressions_check_app","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_impressions_check_app","type":"Service"},{"attrs":{"check_command":"Check DB archive size","display_name":"Check DB archive size","name":"Check DB archive size","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB archive size","type":"Service"},{"attrs":{"check_command":"disk","display_name":"check_disk","name":"check_disk","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_disk","type":"Service"},{"attrs":{"check_command":"check_dataxu_folder","display_name":"check_dataxu_folder","name":"check_dataxu_folder","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_dataxu_folder","type":"Service"},{"attrs":{"check_command":"ssh","display_name":"check_ssh","name":"check_ssh","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_ssh","type":"Service"},{"attrs":{"check_command":"check db connections","display_name":"Check DB connections","name":"Check DB connections","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB connections","type":"Service"},{"attrs":{"check_command":"check_vid_unmap_device","display_name":"check_vid_unmap_device","name":"check_vid_unmap_device","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_vid_unmap_device","type":"Service"},{"attrs":{"check_command":"daily_impressions_check_app_DAILY","display_name":"daily_impressions_check_app_DAILY","name":"daily_impressions_check_app_DAILY","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!daily_impressions_check_app_DAILY","type":"Service"},{"attrs":{"check_command":"ping4","display_name":"ping4","name":"ping4","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!ping4","type":"Service"},{"attrs":{"check_command":"ssh","display_name":"check_ssh","name":"check_ssh","state":0.0},"joins":{},"meta":{},"name":"sftpserver.net!check_ssh","type":"Service"},{"attrs":{"check_command":"disk","display_name":"check_disk","name":"check_disk","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!check_disk","type":"Service"}]}

```

**Comment by danielnelson** [comment]
For clarification, given a check like this the desired output tag would be a `source=appserver.net`?  Taken from `"name": "appserver.net!check_dr_processor_idmissing_reference"`?
```
        {
            "attrs": {
                "check_command": "check_dr_processor_idmissing_reference",
                "display_name": "check_dr_processor_idmissing_reference",
                "name": "check_dr_processor_idmissing_reference",
                "state": 0.0
            },
            "joins": {},
            "meta": {},
            "name": "appserver.net!check_dr_processor_idmissing_reference",
            "type": "Service"
        },
```

**Comment by danielnelson** [comment]
Reading Icinga 2 docs, perhaps we need to pull some additional attrs back.  I don't have an Icinga setup right now, does it work if we add `attrs=host_name` to the above query?
```
curl --user root:abc123 'http://localhost:5665/v1/objects/services?attrs=name&attrs=display_name&attrs=state&attrs=check_command&attrs=host_name'
```

**Comment by buggsdummy** [comment]
Yes that works! It displays the host_name as the affected asset

```
{"results":[{"attrs":{"check_command":"Check DB backup","display_name":"Check DB backup","host_name":"dbserver.net","name":"Check DB backup","state":0.0},"joins":{},"meta":{},"name":"dbserver.net!Check DB backup","type":"Service"},

```

**Comment by buggsdummy** [comment]
Any updates on when this will get pushed?

**Comment by buggsdummy** [comment]
TTT

**Comment by abracadv8** [comment]
@danielnelson  -- Are you accepting pull requests?  What changes are you looking for?

As it stands, the `url` fields are the icinga server itself:
```
	"display_name":  check.Attrs.DisplayName,
	"check_command": check.Attrs.CheckCommand,
	"source":        url.Hostname(),
	"scheme":        url.Scheme,
	"port":          url.Port(),
```

Adding the parameter `&attrs=host_name` gives us the host name of the actual server that needs to be alerted on, but I don't think there's any way to get a port since commandchecks are essentially shell scripts that run on whatever server that needs checks.

I would suggest to simply extend the request as you proposed above and add on an additional field:
```
	"host":          check.Attrs.HostName,
```

This would:

1. Provide the true host where the check is failing
2. Preserve backwards compatibility for anyone currently using those fields

I have already deployed a custom agent with the changes above to our icinga2 server and have been using it the last few weeks.

**Comment by danielnelson** [comment]
Definitely accepting pull requests.

The `host` and `source` tags have special meaning that we try to keep consistent across all plugins:
- `host` is the hostname of the server running Telegraf. It is normally added automatically so we wouldn't want to override it.
- `source` is meant to be the hostname of the system the data is about.  It seems to me this should be set to `check.Attrs.HostName`.

So I think we need a new that will hold the Icinga's servers name.  How about calling this `server` to match the configuration option?

In summary we will have:
```
	"display_name":  check.Attrs.DisplayName,
	"check_command": check.Attrs.CheckCommand,
	"source":        check.Attrs.HostName,
	"server":        url.Hostname(),
	"scheme":        url.Scheme,
	"port":          url.Port(),
```

**Comment by abracadv8** [comment]
Thanks, slightly more confusing, the `services` check requires `host_name` and the 'hosts' check requires `name`.

Services response:

```
{"attrs":{"host_name":"example.host.fqdn.net","name":"procs"},
"joins":{},"meta":{},"name":"hostname!procs","type":"Service"}]}
```

Hosts response:

```
....,"joins":{},"meta":{},"name":"example.host.fqdn.net","type":"Host"}]}
```
 
It doesn't appear that the code uses the `hosts` query though, only `services`: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/icinga2/icinga2.go#L172

So it is fine just to only handle the host_name tag, correct?

**Comment by abracadv8** [comment]
I had to make the changes to handle `services` and `hosts` queries; the unit tests included both so I had to make conditional checks in the request and response parsing to handle the different values.

Can you take a quick look and provide feedback and I will sign the CLA and generate a pull as soon as I get corporate approval: https://github.com/abracadv8/telegraf/commit/187826de4116b2f591e60b9daf7c8957be8c1d15

**Comment by danielnelson** [comment]
Great, can you open a pull request?

**Comment by abracadv8** [comment]
Done -- https://github.com/influxdata/telegraf/pull/7651
